ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/139__trusted_ca_bundles.robot

```diff
@@ -0,0 +1,155 @@
+
+    Log    message=Check that operator is available    level=INFO
+    Is Operator Available    ${RHOAI_OPERATOR_DEPLOYMENT_NAME}    ${RHOAI_OPERATOR_NS}
+    Is DSCI In Ready State    ${DSCI_NAME}    ${RHOAI_OPERATOR_NS}
+
+    Log    message=Validate Trusted CA Bundle Management state Managed    level=INFO
+
+    Create Namespace In Openshift    ${TEST_NS}
+    Wait Until Keyword Succeeds    3 min    0 sec
+    ...    Is Resource Present    project    ${TEST_NS}    ${TEST_NS}    ${IS_PRESENT}
+
+    Wait Until Keyword Succeeds    3 min    0 sec
+    ...    Is Resource Present     ConfigMap    ${TRUSTED_CA_BUNDLE_CONFIGMAP}    ${TEST_NS}    ${IS_PRESENT}
+
+    # Check that ConfigMap contains "ca-bundle.crt"
+    Wait Until Keyword Succeeds    3 min    0 sec
+    ...    Check ConfigMap Contains CA Bundle   ${TRUSTED_CA_BUNDLE_CONFIGMAP}    ca-bundle.crt    ${TEST_NS}
+
+    Set Custom CA Bundle Value In DSCI   ${CUSTOM_CA_BUNDLE}    ${RHOAI_OPERATOR_NS}
+    Wait Until Keyword Succeeds    3 min    0 sec
+    ...    Is CA Bundle Value Present    ${TRUSTED_CA_BUNDLE_CONFIGMAP}    ${CUSTOM_CA_BUNDLE}    ${TEST_NS}    ${IS_PRESENT}
+
+    Log    message=Validate Trusted CA Bundle Management State Unmanaged    level=INFO
+
+    Set Trusted CA Bundle Management State    Unmanaged    ${RHOAI_OPERATOR_NS}
+
+    # Trusted CA BUndle managementStatus 'Unmanaged' should NOT result in bundle being overwirtten by operator
+    Set Custom CA Bundle Value On ConfigMap
+    ...    ${TRUSTED_CA_BUNDLE_CONFIGMAP}    random-ca-bundle-value    ${TEST_NS}
+    # Allow operator time to reconsile
+    Sleep    5
+    Wait Until Keyword Succeeds    3 min    0 sec
+    ...    Is CA Bundle Value Present    ${TRUSTED_CA_BUNDLE_CONFIGMAP}    random-ca-bundle-value    ${TEST_NS}    ${IS_PRESENT}
+
+    Log    message=Validate Trusted CA Bundle Management State Removed    level=INFO
+
+    Set Trusted CA Bundle Management State    Removed    ${RHOAI_OPERATOR_NS}
+
+    # Check that odh-trusted-ca-bundle has been 'Removed'
+    Wait Until Keyword Succeeds    3 min    0 sec
+    ...    Is Resource Present     ConfigMap    ${TRUSTED_CA_BUNDLE_CONFIGMAP}    ${TEST_NS}    ${IS_NOT_PRESENT}
+
+    Log    message=Restore DSCI to original state    level=INFO
+    Set Custom CA Bundle Value In DSCI   ''    ${RHOAI_OPERATOR_NS}
+    Set Trusted CA Bundle Management State    Managed    ${RHOAI_OPERATOR_NS}
+    Delete Namespace From Openshift    ${TEST_NS}
+
+
+*** Keywords ***
+Suite Setup
+    [Documentation]    Suite Setup
+    Log    message=Suite Setup.    level=INFO
+    RHOSi Setup
+
+Suite Teardown
+    [Documentation]    Suite Teardown
+    Log    message=Suite Teardown.    level=INFO
+    RHOSi Teardown
+
+Is Operator Available
+    [Documentation]    Checks if operator is available
+    [Arguments]    ${operator_name}    ${namespace}
+    ${rc}=    Run And Return Rc
+    ...    oc wait --timeout=2m --for condition=available -n ${namespace} deploy/${operator_name}
```

Comment:
maybe the keyword name should be sth like `Wait until operator is ready/running/whatever`